service/vpn_key_rotation_webhook_validation.go

```diff
@@ -16,3 +19,26 @@ func ValidateVpnKeyRotationCreate(ctx context.Context, r *controllerv1alpha1.Vpn
 	}
 	return nil
 }
+
+func ValidateVpnKeyRotationDelete(ctx context.Context, r *controllerv1alpha1.VpnKeyRotation) error {
+	slice := &controllerv1alpha1.SliceConfig{}
+	found, err := util.GetResourceIfExist(ctx, client.ObjectKey{
+		Name:      r.Spec.SliceName,
+		Namespace: r.Namespace,
+	}, slice)
+	if err != nil {
+		return err
+	}
+	if found && slice.ObjectMeta.DeletionTimestamp.IsZero() {
+		//Load Event Recorder with project name, vpnkeyrotation(slice) name and namespace
+		eventRecorder := util.CtxEventRecorder(ctx).
+			WithProject(util.GetProjectName(r.Namespace)).
+			WithNamespace(r.Namespace).
+			WithSlice(r.Name)
+		//Register an event for worker slice config deleted forcefully
+		util.RecordEvent(ctx, eventRecorder, r, slice, events.EventIllegalVPNKeyRotationConfigDelete)
+		return fmt.Errorf("sliceconfig %s not allowed to delete unless slice is deleted", r.Name)
```

Comment:
Can we change this message a bit. 
`VPN Key is not allowed to delete unless sliceconfig is deleted.`